Re-enable some Quantization UTs after Quantization flow updates

leslie-fang-intel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #108131
* __->__ #108125

**Summary**
This diff mainly has done 2 things:

- Re-enable the testcases skipped in commit: https://github.com/pytorch/pytorch/commit/9ae3d7ca90ebeaea1082f41f4c60ee3488954d78 due to the quantization flow update.
- Break down the original testcases into small testcases to make each testcase simpler.

**TestPlans**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_relu
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_add
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_add_relu
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_dequant_promotion
python -m pytest test_mkldnn_pattern_matcher.py -k test_qlinear
python -m pytest test_mkldnn_pattern_matcher.py -k test_qlinear_relu
python -m pytest test_mkldnn_pattern_matcher.py -k test_qlinear_dequant_promotion
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

leslie-fang-intel committed:
Re-enable the quantization UT after Quantization flow updates

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "Re-enable some Quantization UTs after Quantization flow updates"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "Re-enable some Quantization UTs after Quantization flow updates"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "Re-enable some Quantization UTs after Quantization flow updates"


**Summary**
This diff mainly has done 2 things:

- Re-enable the testcases skipped in commit: https://github.com/pytorch/pytorch/commit/9ae3d7ca90ebeaea1082f41f4c60ee3488954d78 due to the quantization flow update.
- Break down the original testcases into small testcases to make each testcase simpler.

**TestPlans**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_relu
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_add
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_add_relu
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_dequant_promotion
python -m pytest test_mkldnn_pattern_matcher.py -k test_qlinear
python -m pytest test_mkldnn_pattern_matcher.py -k test_qlinear_relu
python -m pytest test_mkldnn_pattern_matcher.py -k test_qlinear_dequant_promotion
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

leslie-fang-intel commented:
Hi @jerryzh168, as we synced in slack, I have re-enable and break down the testcases. Please kindly help to take a look again.

jerryzh168 reviewed (approved):
thanks, I'm wondering if it is possible to break down to even simpler patterns? also can you add a comment for each test to draw the pattern you are quantizing as well, that might make it clearer

leslie-fang-intel committed:
Update on "Re-enable some Quantization UTs after Quantization flow updates"


**Summary**
This diff mainly has done 2 things:

- Re-enable the testcases skipped in commit: https://github.com/pytorch/pytorch/commit/9ae3d7ca90ebeaea1082f41f4c60ee3488954d78 due to the quantization flow update.
- Break down the original testcases into small testcases to make each testcase simpler.

**TestPlans**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_relu
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_add
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_add_relu
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_dequant_promotion
python -m pytest test_mkldnn_pattern_matcher.py -k test_qlinear
python -m pytest test_mkldnn_pattern_matcher.py -k test_qlinear_relu
python -m pytest test_mkldnn_pattern_matcher.py -k test_qlinear_dequant_promotion
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "Re-enable some Quantization UTs after Quantization flow updates"


**Summary**
This diff mainly has done 2 things:

- Re-enable the testcases skipped in commit: https://github.com/pytorch/pytorch/commit/9ae3d7ca90ebeaea1082f41f4c60ee3488954d78 due to the quantization flow update.
- Break down the original testcases into small testcases to make each testcase simpler.

**TestPlans**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_relu
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_add
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_add_relu
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_dequant_promotion
python -m pytest test_mkldnn_pattern_matcher.py -k test_qlinear
python -m pytest test_mkldnn_pattern_matcher.py -k test_qlinear_relu
python -m pytest test_mkldnn_pattern_matcher.py -k test_qlinear_dequant_promotion
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "Re-enable some Quantization UTs after Quantization flow updates"


**Summary**
This diff mainly has done 2 things:

- Re-enable the testcases skipped in commit: https://github.com/pytorch/pytorch/commit/9ae3d7ca90ebeaea1082f41f4c60ee3488954d78 due to the quantization flow update.
- Break down the original testcases into small testcases to make each testcase simpler.

**TestPlans**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_relu
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_add
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_add_relu
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_dequant_promotion
python -m pytest test_mkldnn_pattern_matcher.py -k test_qlinear
python -m pytest test_mkldnn_pattern_matcher.py -k test_qlinear_relu
python -m pytest test_mkldnn_pattern_matcher.py -k test_qlinear_dequant_promotion
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

leslie-fang-intel commented:
@pytorchbot merge
